src/shared/tests.h

```diff
@@ -201,41 +201,47 @@ static inline int run_test_table(void) {
-                if (_result < 0) {                                                                       \
-                        log_error_errno("Assertion failed: %s (result: %d, error: %m)", #expr, _result); \
-                        abort();                                                                         \
-                }                                                                                        \
+#define ASSERT_OK(expr)                                                                                          \
+        ({                                                                                                       \
+                typeof(expr) _result = (expr);                                                                   \
+                if (_result < 0) {                                                                               \
+                        log_error_errno(_result, "%s:%i: Unexpected error: %m", PROJECT_FILE, __LINE__);         \
+                        abort();                                                                                 \
+                }                                                                                                \
         })
 
-#define ASSERT_EQ(expr1, expr2)                                                                          \
-        ({                                                                                               \
-                int _expr1 = (expr1);                                                                    \
-                int _expr2 = (expr2);                                                                    \
-                if (_expr1 != _expr2) {                                                                  \
-                        log_error("Assertion failed: expected %s == %s, but %d != %d", #expr1, #expr2, _expr1, _expr2); \
-                        abort();                                                                         \
-                }                                                                                        \
+#define ASSERT_EQ(expr1, expr2)                                                                                  \
+        ({                                                                                                       \
+                typeof(expr1) _expr1 = (expr1);                                                                  \
+                typeof(expr2) _expr2 = (expr2);                                                                  \
+                if (_expr1 != _expr2) {                                                                          \
+                        log_error("%s:%i: Assertion failed: "                                                    \
+                                  "expected %s == %s, but %d != %d",                                             \
```

Comment:
`%d` only works for integers.

Please take a look at https://stackoverflow.com/questions/25445014/typeof-variables-and-printf and use that to use the correct printf format specifier depending on the type of the input expressions.

(This applies to all the macros)

Reply:
@DaanDeMeyer Thanks for the link. Trust me, I've been on this since I got the notification. I was encountering some errors that I had to debug. At one point, I was going to ask for help, but I decided to work on it more until I was able to resolve the error. It was interesting and educational debugging the error.